wp-parsely.php

```diff
@@ -67,6 +62,51 @@ public static function set_version( string $version ): void {
 		self::$version = $version;
 	}
 
+	/**
+	 * Returns configuration which represents how the plugin is configured in site.
+	 *
+	 * @return ParselyConfigs
+	 */
+	public static function get_configs() {
+		if ( ! self::is_active() ) {
+			return null;
+		}
+
+		/**
+		 * Variable.
+		 *
+		 * @var ParselyConfigs
+		 */
+		$configs = new stdClass();
+		$options = self::get_parsely_options() ?: [];
+
+		$configs->is_pinned_version            = has_filter( 'wpvip_parsely_version' );
+		$configs->site_id                      = $options['apikey'] ?? '';
+		$configs->have_api_secret              = '' !== ( $options['api_secret'] ?? '' );
+		$configs->is_javascript_disabled       = (bool) ( $options['disable_javascript'] ?? false );
+		$configs->is_autotracking_disabled     = (bool) ( $options['disable_autotrack'] ?? false );
+		$configs->should_track_logged_in_users = (bool) ( $options['track_authenticated_users'] ?? false );
+
+		$configs->tracked_post_types = array();
+		$post_names                  = get_post_types( array( 'public' => true ) );
+		foreach ( $post_names as $post_name ) {
+			$tracked_post_type       = new stdClass();
```

Comment:
I think it'll be a bit more on par with the rest of the codebase to use an array here instead of an object.

```
$tracked_post_type = [ 'name'  => $post_name ];
...
$tracked_post_type['track_type'] = 'post';
```

Reply:
I found this as a way use PHPDoc to properly document the returned array but still using `stdClass` seems better because of better autocompletion in code editors.

```
	/**
	 * Returns configuration which represents how the plugin is configured in site.
	 *
	 * @return array{
	 *   is_pinned_version: bool,
	 *   site_id: string,
	 *   have_api_secret: bool,
	 *   is_javascript_disabled: bool,
	 *   is_autotracking_disabled: bool,
	 *   should_track_logged_in_users: bool,
	 *   tracked_post_types: array{
	 *     name: string,
	 *     track_type: string,
	 *   }[]
	 * }
	 */
	public static function get_configs()
```

Let me know which method do you think is most appropriate and I will implement that. Thanks

Reply:
@Automattic/vip-platform-cantina curious to hear the team's thoughts on this, including the `types/wp-parsely.php` file.

Reply:
I'm leaning towards the array docblock to be honest, at least for this use case. PHP in particular is moving away from dynamically added class properties as well. And a full fledged class seems kinda overkill here.

I'm all for better typing (pretty sure we support Psalm already), but I think there's a larger conversation to be had and I'd rather not introduce a one-off pattern with the `types/wp-parsely.php` thing.